src/material-experimental/mdc-slider/slider.ts

```diff
@@ -949,18 +957,18 @@ export class MatSlider
 
     // MDC only updates the slider when the window is resized which
     // doesn't capture changes of the container itself. We use a resize
-    // observer to ensure that the layout is correct (see #24590).
+    // observer to ensure that the layout is correct (see #24590 and #25286).
     this._ngZone.runOutsideAngular(() => {
-      // The callback will fire as soon as an element is observed and
-      // we only want to know after the initial layout.
-      let hasResized = false;
-      this._resizeObserver = new ResizeObserver(() => {
-        if (hasResized) {
-          // Debounce the layouts since they can happen frequently.
-          clearTimeout(this._resizeTimer);
-          this._resizeTimer = setTimeout(this._layout, 50);
-        }
-        hasResized = true;
+      this._resizeObserver = new ResizeObserver(entries => {
+        clearTimeout(this._resizeTimer);
+        this._resizeTimer = setTimeout(() => {
+          // The `layout` call is going to call `getBoundingClientRect` to update the dimensions
+          // of the host. Since the `ResizeObserver` already calculated them, we can save some
+          // work by returning them instead of having to check the DOM again.
```

Comment:
Optional nit: ```suggestion
          // work by caching them instead of having to check the DOM again.
```

Reply:
The caching isn't permanent though. It gets reset after the `layout` call.

Reply:
I see. I just wasn't sure where you "returned" it, but the comment makes sense regardless.